src/components/screens/CommunityScreen/CommunityHero.js

```diff
@@ -149,8 +151,42 @@ const MailingListWrapper = styled.div`
   }
 `;
 
-export default function CommunityHero(props) {
+const fetchNpmDownloads = async () => {
+  const promises = Object.values(url.npmApi).map(async (uri) => {
+    const response = await fetch(uri);
+    const json = await response.json();
+
+    return json.downloads;
+  });
+
+  const results = await Promise.all(promises);
+
+  return results.reduce((a, b) => a + b, 0);
+};
+
+export const CommunityHero = ({ npmDownloads, updateNpmDownloads, ...props }) => {
   const [namespace, repo] = url.gitHub.repo.match(/github.com\/(.*)\/(.*)$/).slice(1);
+
+  if (typeof window !== `undefined`) {
```

Comment:
Would it make sense to pull this logic out into an `async getNpmDownloads` function, and call it from a `lifecycle({ componentDidMount: ... })` `recompose` HoC below? We could abstract the combination of `withState`, `withHandlers` and `lifecycle` into a single `withNpmDownloads` HoC maybe?

Let me know if you want me to expand further

Reply:
Resolved in last commit. Hope that it was right, i'm new to react ;)